### PR TITLE
Replace errors.Wrap with fmt.Errorf

### LINKED ISSUE #124: Replace errors.Wrap with fmt.Errorf from go 1.13
## Is your feature request related to a problem? Please describe

Go has introduced error wrapping with `fmt.Errorf`. We should remove the wrap methods from the errors package.

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

This PR closes #124

## Short description of the changes

- Deleted errors package functions and replaced with fmt.Errorf
- Remove errors package
- Update requirement for Patron to be Go 1.13+

### CODE REVIEW

**Review by drakos74 — DISMISSED** [review]

File: async/component.go [hunk 1]
```diff
@@ -152,16 +154,20 @@ func (c *Component) executeFailureStrategy(msg Message, err error) error {
 	log.FromContext(msg.Context()).Errorf("failed to process message, failure strategy executed: %v", err)
 	switch c.failStrategy {
 	case NackExitStrategy:
-		return errors.Aggregate(err, errors.Wrap(msg.Nack(), "failed to NACK message"))
+		err1 := msg.Nack()
```
Comment (fpapadopou) [inline async/component.go]: How about `nackErr`instead of `err1`?

**Review by fpapadopou — CHANGES_REQUESTED** [review]

**Review by fpapadopou — DISMISSED** [review]

**Comment by fpapadopou** [comment]
@mantzas will you resolve conflicts? I can do it too, if you want

**Comment by mantzas** [comment]
> @mantzas will you resolve conflicts? I can do it too, if you want

@fpapadopou that's the big fun... i am on it! Thanks

File: async/component_test.go [hunk 2]
```diff
@@ -3,7 +3,10 @@ package async
 import (
 	"context"
 	"errors"
+<<<<<<< HEAD
```
Comment (fpapadopou) [inline async/component_test.go]: Oops

**Review by fpapadopou — CHANGES_REQUESTED** [review]

File: async/component_test.go (hunk 2 above)
Comment (mantzas) [reply async/component_test.go]: Seen it... on it

**Review by fpapadopou — APPROVED** [review]

**Review by drakos74 — APPROVED** [review]

**Pull request merged**